src/main/scala/org/embulk/output/s3_parquet/parquet/LogicalTypeHandler.scala

```diff
@@ -0,0 +1,58 @@
+package org.embulk.output.s3_parquet.parquet
+
+import org.apache.parquet.io.api.{Binary, RecordConsumer}
+import org.apache.parquet.schema.PrimitiveType.PrimitiveTypeName
+import org.apache.parquet.schema.{OriginalType, PrimitiveType, Type}
+import org.embulk.spi.time.Timestamp
+import org.msgpack.value.Value
+
+/**
+ * Handle Apache Parquet 'Logical Types' on schema/value conversion.
+ * ref. https://github.com/apache/parquet-format/blob/master/LogicalTypes.md
+ *
+ * It focuses on only older representation because newer supported since 1.11 is not used actually yet.
+ * TODO Support both of older and newer representation after 1.11+ is published and other middleware supports it.
+ *
+ */
+trait LogicalTypeHandler {
+    def newSchemaFieldType(name: String): PrimitiveType
+
+    def consume(orig: AnyRef, recordConsumer: RecordConsumer): Unit
+}
```

Comment:
In this case, I think that you do not need to use `AnyRef`. How is the below code?

```
trait LogicalTypeHandler[A] {
    def newSchemaFieldType(name: String): PrimitiveType

    def consume(orig: A, recordConsumer: RecordConsumer): Unit
}
```

Please let me know if there is a case you want to care with this code.

Reply:
Oh sorry, I got this `AnyRef` meaning. This `AnyRef` is to consume unknown data in the `ColumnVisitor`, right? OK, there is no help for it.

Reply:
that's right ... 😭